src/reference/antora/modules/ROOT/pages/jdbc/message-store-json.adoc

```diff
@@ -0,0 +1,262 @@
+}
+----
+
+The string parameter (`"com.example"`) specifies additional trusted packages for deserialization.
+These packages are appended to the default trusted packages (see <<trusted-packages>> section).
+Only classes from trusted packages can be deserialized for security.
+
+== Database Schema Modification
+
+[IMPORTANT]
+====
+JSON serialization **requires modifying the database schema**.
+The default schema with BLOB/BYTEA column types cannot be used for JSON serialization.
+====
+
+The `MESSAGE_CONTENT` column must be changed to a text-based type that can store JSON.
+
+=== PostgreSQL
+
+For PostgreSQL, the `JSONB` type can be used.
+
+[source,sql]
+----
+-- JSONB (enables JSON queries)
+ALTER TABLE INT_CHANNEL_MESSAGE
+ALTER COLUMN MESSAGE_CONTENT TYPE JSONB
+USING MESSAGE_CONTENT::text::jsonb;
+----
+
+=== MySQL
+
+For MySQL, the `JSON` type can be used.
+
+[source,sql]
+----
+-- JSON type (enables JSON functions)
+ALTER TABLE INT_CHANNEL_MESSAGE
+MODIFY COLUMN MESSAGE_CONTENT JSON;
+----
+
+=== H2
+
+For H2 database, the `CLOB` type can be used.
+
+[source,sql]
+----
+ALTER TABLE INT_CHANNEL_MESSAGE
+ALTER COLUMN MESSAGE_CONTENT CLOB;
+----
+
+=== Other Databases
+
+For any database that supports large text columns (CLOB, TEXT, etc.), the `MESSAGE_CONTENT` column can be modified to an appropriate text type.
+
+== Example Schema for JSON Serialization
+
+The following examples demonstrate how to create a dedicated table for JSON-based message storage.
+
+=== PostgreSQL
+
+[source,sql]
+----
+CREATE TABLE JSON_CHANNEL_MESSAGE (
+   MESSAGE_ID CHAR(36) NOT NULL,
+   GROUP_KEY CHAR(36) NOT NULL,
+   CREATED_DATE BIGINT NOT NULL,
+   MESSAGE_PRIORITY BIGINT,
+   MESSAGE_SEQUENCE BIGINT NOT NULL DEFAULT nextval('JSON_MESSAGE_SEQ'),
+   MESSAGE_CONTENT JSONB, -- JSON message content
+   REGION VARCHAR(100) NOT NULL,
+   CONSTRAINT JSON_CHANNEL_MESSAGE_PK
+      PRIMARY KEY (REGION, GROUP_KEY, CREATED_DATE, MESSAGE_SEQUENCE)
+);
+----
+
+=== MySQL
+
+[source,sql]
+----
+CREATE TABLE JSON_CHANNEL_MESSAGE (
+   MESSAGE_ID CHAR(36) NOT NULL,
+   GROUP_KEY CHAR(36) NOT NULL,
+   CREATED_DATE BIGINT NOT NULL,
+   MESSAGE_PRIORITY BIGINT,
+   MESSAGE_SEQUENCE BIGINT NOT NULL AUTO_INCREMENT UNIQUE,
+   MESSAGE_CONTENT JSON, -- JSON message content
+   REGION VARCHAR(100) NOT NULL,
+   CONSTRAINT JSON_CHANNEL_MESSAGE_PK
+      PRIMARY KEY (REGION, GROUP_KEY, CREATED_DATE, MESSAGE_SEQUENCE)
+);
+----
+
+=== H2
+
+[source,sql]
+----
+CREATE TABLE JSON_CHANNEL_MESSAGE (
+   MESSAGE_ID CHAR(36) NOT NULL,
+   GROUP_KEY CHAR(36) NOT NULL,
+   CREATED_DATE BIGINT NOT NULL,
+   MESSAGE_PRIORITY BIGINT,
+   MESSAGE_SEQUENCE BIGINT NOT NULL DEFAULT NEXT VALUE FOR JSON_MESSAGE_SEQ,
+   MESSAGE_CONTENT CLOB, -- JSON message content
+   REGION VARCHAR(100) NOT NULL,
+   CONSTRAINT JSON_CHANNEL_MESSAGE_PK
+      PRIMARY KEY (REGION, GROUP_KEY, CREATED_DATE, MESSAGE_SEQUENCE)
+);
+----
+
+== JSON Structure
+
+Messages are stored with the following JSON structure:
```

Comment:
I think we need to emphasize here that this is a result of Jackson polymorphic serialization.